packages/cli/commands/charm.ts

```diff
@@ -143,7 +144,62 @@ export const charm = new Command()
   .arguments("<entry:string>")
   .action((options, entryPath) =>
     setCharmRecipe(parseCharmOptions(options), absPath(entryPath))
-  );
+  )
+  /* charm link */
+  .command("link", "Link a field from one charm to another")
+  .usage(spaceUsage)
+  .example(
+    `ct charm link ${EX_ID} ${EX_COMP} bafycharm1/outputEmails bafycharm2/emails`,
+    `Link outputEmails field from charm "bafycharm1" to emails field in charm "bafycharm2".`,
+  )
+  .example(
+    `ct charm link ${EX_ID} ${EX_COMP} bafycharm1/data/users/0/email bafycharm2/config/primaryEmail`,
+    `Link deep nested field including array access.`,
+  )
+  .arguments("<source:string> <target:string>")
+  .action(async (options, sourceRef, targetRef) => {
+    const spaceConfig = parseSpaceOptions(options);
+
+    // Parse source reference (charmId/path/to/field)
+    const sourceParts = sourceRef.split("/");
+    if (sourceParts.length < 2) {
+      throw new ValidationError(
+        `Invalid source reference format. Expected: charmId/path/to/field`,
+        { exitCode: 1 },
+      );
+    }
+    const sourceCharmId = sourceParts[0];
+    const sourcePath = sourceParts.slice(1).map((segment) => {
```

Comment:
nit: Could be `parseLink(ref: string): { charmId: string, path?: string }` and can use that directly in the unit tests, like the `parse[Charm|Space]Options()` functions here (these `command/` files are almost all parsing inputs)